plugins/woocommerce/client/admin/client/marketing/components/PluginCardBody/SmartPluginCardBody.tsx

```diff
@@ -40,8 +39,18 @@ export const SmartPluginCardBody = ( {
 		null
 	);
 	const { installAndActivatePlugins } = useDispatch( PLUGINS_STORE_NAME );
-	const { isPluginInstalledNotActivated } =
-		useIsPluginInstalledNotActivated();
+	const { installState } = useSelect(
```

Comment:
This seems to not only be about TS issues. I'm not sure how to test these changes. Do you have any advice?

Reply:
I've already found
<img width="1141" alt="image" src="https://github.com/user-attachments/assets/a75a24c9-dd82-42b7-9c81-548cdf890df5" />

Reply:
minor suggestion: instead of returning an object, why not simply return the `initialState` string?

```es6
	const installState = useSelect(
		( select ) => {
			const { getPluginInstallState } = select(
				PLUGINS_STORE_NAME
			) as PluginSelectors;

			return getPluginInstallState( plugin.product );
		},
		[ plugin.product ]
	);
```

Reply:
> This seems to not only be about TS issues. 

Oh yes, I did a bit of refactor and simplified the code. The code was previously in `useIsPluginInstalledNotActivated`, which is now deleted. See [`36e0012` (#54485)](https://github.com/woocommerce/woocommerce/pull/54485/commits/36e0012e8df35d5d1697c1a846c1b8999a12c972).

> minor suggestion: instead of returning an object, why not simply return the `initialState` string?

Technically that would work, but it is better to always return objects. See [Best practices for using useSelect() from @wordpress/data](https://jardasn.blog/2021/01/22/some-best-practices-for-using-useselect-from-wordpress-data/) (look for "Prefer returning objects with properties from the callback").

Reply:
:til: Thanks a lot for the explanation and the reference!